tests/subsys/pm/power_mgmt/src/main.c

```diff
@@ -366,6 +379,30 @@ void test_device_order(void)
 	testing_device_order = false;
 }
 
+void test_device_constraint(void)
+{
+	int ret;
+
+	pm_device_runtime_enable(device_a);
+	(void)pm_device_runtime_get(device_a);
+
+	pm_device_constraint_set((struct device *)device_a);
+	zassert_true(pm_device_constraint_check(device_a), NULL);
+
+	ret = pm_device_runtime_put(device_a);
```

Comment:
runtime API behavior should be tested in the runtime API test

Reply:
yep, had put here because could check it in both scnearios, system and runtime pm. Since you are removing runtime pm from this test I'll move it as well.